editor/components/url-input/index.js

```diff
@@ -25,18 +25,41 @@ const stopEventPropagation = ( event ) => event.stopPropagation();
 class UrlInput extends Component {
 	constructor() {
 		super( ...arguments );
+
 		this.onChange = this.onChange.bind( this );
 		this.onKeyDown = this.onKeyDown.bind( this );
 		this.bindListNode = this.bindListNode.bind( this );
 		this.updateSuggestions = throttle( this.updateSuggestions.bind( this ), 200 );
+
 		this.suggestionNodes = [];
+
 		this.state = {
 			posts: [],
 			showSuggestions: false,
 			selectedSuggestion: null,
 		};
 	}
 
+	componentDidUpdate() {
+		const { showSuggestions, selectedSuggestion } = this.state;
+		// only have to worry about scrolling selected suggestion into view
+		// when already expanded
+		if ( showSuggestions && selectedSuggestion !== null && ! this.scrollingIntoView ) {
+			this.scrollingIntoView = true;
+			scrollIntoView( this.suggestionNodes[ selectedSuggestion ], this.listNode, {
+				onlyScrollIfNeeded: true,
+			} );
+
+			setTimeout( () => {
```

Comment:
Unrelated, we should probably replace this with `withSafeTimeout` :)

Reply:
We should probably not have `setTimeout` of any form 😉